docs/source/tasks/asr.mdx

```diff
@@ -0,0 +1,222 @@
+
+>>> timit = load_dataset("timit_asr")
+```
+
+Then take a look at an example:
+
+```py
+>>> timit
+DatasetDict({
+    train: Dataset({
+        features: ['file', 'audio', 'text', 'phonetic_detail', 'word_detail', 'dialect_region', 'sentence_type', 'speaker_id', 'id'],
+        num_rows: 4620
+    })
+    test: Dataset({
+        features: ['file', 'audio', 'text', 'phonetic_detail', 'word_detail', 'dialect_region', 'sentence_type', 'speaker_id', 'id'],
+        num_rows: 1680
+    })
+})
+```
+
+While the dataset contains a lot of helpful information, like `dialect_region` and `sentence_type`, you will focus on the `audio` and `text` fields in this guide. Remove the other columns:
+
+```py
+>>> timit = timit.remove_columns(
+...     ["phonetic_detail", "word_detail", "dialect_region", "id", "sentence_type", "speaker_id"]
+... )
+```
+
+Take a look at the example again:
+
+```py
+>>> timit["train"][0]
+{'audio': {'array': array([-2.1362305e-04,  6.1035156e-05,  3.0517578e-05, ...,
+         -3.0517578e-05, -9.1552734e-05, -6.1035156e-05], dtype=float32),
+  'path': '/root/.cache/huggingface/datasets/downloads/extracted/404950a46da14eac65eb4e2a8317b1372fb3971d980d91d5d5b221275b1fd7e0/data/TRAIN/DR4/MMDM0/SI681.WAV',
+  'sampling_rate': 16000},
+ 'file': '/root/.cache/huggingface/datasets/downloads/extracted/404950a46da14eac65eb4e2a8317b1372fb3971d980d91d5d5b221275b1fd7e0/data/TRAIN/DR4/MMDM0/SI681.WAV',
+ 'text': 'Would such an act of refusal be useful?'}
+```
+
+The `audio` column contains a 1-dimensional `array` of the speech signal that must be called to load and resample the audio file.
+
+## Preprocess
+
+Load the Wav2Vec2 processor to process the audio signal and transcribed text:
+
+```py
+>>> from transformers import AutoProcessor
+
+>>> processor = AutoProcessor.from_pretrained("facebook/wav2vec2-base")
+```
+
+The preprocessing function needs to:
+
+1. Call the `audio` column to load and resample the audio file.
+2. Extract the `input_values` from the audio file.
+2. Typically, when you call the processor, you call the feature extractor. Since you also want to tokenize text, instruct the processor to call the tokenizer instead with a context manager.
+
+```py
+>>> def prepare_dataset(batch):
+...     audio = batch["audio"]
+    
+...     batch["input_values"] = processor(audio["array"], sampling_rate=16000).input_values[0]
+...     batch["input_length"] = len(batch["input_values"])
+
+...     with processor.as_target_processor():
+...          batch["labels"] = processor(batch["text"]).input_ids
+...     return batch
+```
+
+Use 🤗 Datasets [`map`](https://huggingface.co/docs/datasets/package_reference/main_classes.html#datasets.Dataset.map) function to apply the preprocessing function over the entire dataset. You can speed up the map function by setting `batched=True` to process multiple elements of the dataset at once and increasing the number of processes with `num_proc`. Remove the columns you don't need:
```

Comment:
```suggestion
Use 🤗 Datasets [`map`](https://huggingface.co/docs/datasets/package_reference/main_classes.html#datasets.Dataset.map) function to apply the preprocessing function over the entire dataset. You can increasing the number of processes with `num_proc`. Remove the columns you don't need:
```

This doesn't work well because batched audio loading / sampling is done in a loop currently under the hood no @lhoestq ?